### PR TITLE
feat(web-iconics):  implement teams page

### PR DESCRIPTION
**Pull request by zerihunMoges**
Screenshot 2023-05-04 000234

implement teams page,
remove the empty teams page file
change img to next image

### CODE REVIEW

**Review by sam5235 — DISMISSED** [review]

**Review by sam5235 — APPROVED** [review]

File: Web/starter_project_iconics/components/team/TeamCard.tsx
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/TeamCard.tsx:1]: When you define a component, make sure you use react.FC.

File: Web/starter_project_iconics/pages/teams/index.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import TeamsHero from '@/components/team/TeamsHero'
+import Teams from '@/components/team/Teams'
+
 
 function TeamsPage() {
   return (
-    <TeamsHero></TeamsHero>
+    <Teams></Teams>
```
Comment (ruthgdb) [inline Web/starter_project_iconics/pages/teams/index.tsx:6]: If you're not going to add children you dont need closing tags, you can just use ```<Teams />```

File: Web/starter_project_iconics/pages/teams/index.tsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import TeamsHero from '@/components/team/TeamsHero'
+import Teams from '@/components/team/Teams'
+
 
 function TeamsPage() {
```
Comment (ruthgdb) [inline Web/starter_project_iconics/pages/teams/index.tsx:4]: Use react.FC here as well

File: Web/starter_project_iconics/components/team/Teams.tsx
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/Teams.tsx:1]: This component can also be the index file on the teams folder. If you're just calling the other components here and nothing else, you dont need this component.

File: Web/starter_project_iconics/components/team/TeamsHero.tsx [hunk 3]
```diff
@@ -1,22 +1,23 @@
+import Image from "next/image"
 
 function TeamsHero() {
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/TeamsHero.tsx:3]: Use react.FC here

File: Web/starter_project_iconics/components/team/TeamsHero.tsx [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
                     </h1>
                     <p className="text-2xl text-black m-2">Meet our development team is a small but highly skilled and experienced group of professionals. We have a talented mix of web developers, software engineers, project managers and quality assurance specialists who are passionate about developing exceptional products and services.</p>
                 </div>
                 
                 <div className=" w-[85%] md:w-[85%] lg:w-[100%] relative p-8 m-auto">
-                    <img src="\img\teams\team-work\team-work-1.png" alt="" className='contain absolute top-0 left-0 w-[35%]'/>
-                    <img src="\img\teams\team-work\team-work-2.png" alt="" className='contain absolute top-0 right-0 w-[35%]'/>
-                    <img src="\img\teams\team-work\team-work-3.png" alt="" className='contain absolute bottom-0 left-0 w-[35%]'/>
-                    <img src=".\img\teams\hero-background.png" alt="" className="src" />
+                    <Image src="/img/teams/team-work/team-work-1.png" alt="" width={322} height={316.7} className='contain absolute top-0 left-0 w-[35%]'/>
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/TeamsHero.tsx:17]: You can store this images in an array and loop over them. Most of these seem redundant.

File: Web/starter_project_iconics/components/team/Teams.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    const teams = teamMembers
+
+    return (
+        <div className='flex'>
+            <div className='max-w-screen-2xl mx-auto'>
+            <TeamsHero></TeamsHero>
+            
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3  justify-items-center items-center gap-0 p-16">
+                {teams.map((person, index) => <TeamCard key={index} name={person.name} description={person.description} job={person.job} avatar={person.avatar} socialMedia={person.socialMedia}></TeamCard>)}
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/Teams.tsx:15]: You can just pass the whole team member object to the component. 

```
{teamMembers.map((teamMember, index) => <TeamCard key={index} teamMember={teamMember} /> )}
```

File: Web/starter_project_iconics/components/team/Teams.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { teamMembers } from '@/data/team/teams'
+import TeamCard from './TeamCard'
+import TeamsHero from './TeamsHero'
+
+function Teams() {
+
+    const teams = teamMembers
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/Teams.tsx:7]: What is the need for this

File: Web/starter_project_iconics/components/team/Teams.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    return (
+        <div className='flex'>
+            <div className='max-w-screen-2xl mx-auto'>
+            <TeamsHero></TeamsHero>
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/team/Teams.tsx:12]: You dont need a closing tag here as well

**Review by ruthgdb — CHANGES_REQUESTED** [review]
This looks nice overall. You need to address this comments before we can merge this.

File: Web/starter_project_iconics/components/teams/TeamMembersCard.tsx
Comment (ruthgdb) [inline Web/starter_project_iconics/components/teams/TeamMembersCard.tsx:1]: [ver-nit] this is a single team member card so it shouldnt be plural. Change this file and the component inside it to `TeamMemberCard`

File: Web/starter_project_iconics/components/teams/TeamMembersCard.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        className="rounded-full w-32 h-32 mt-2 bg-gray-200 object-contain"
+        src={teamMember.avatar}
+        alt="avatar"
+        width={178}
+        height={178}
+      ></Image>
+
+      <h1 className="font-bold uppercase text-black text-2xl m-3">{teamMember.name}</h1>
+      <h2 className="uppercase text-black text-xl">{teamMember.job}</h2>
+      <p className="text-center my-4 text-xl text-[#7D7D7D]">{teamMember.description}</p>
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/teams/TeamMembersCard.tsx:34]: You can use the secondary color on your theme here

File: Web/starter_project_iconics/components/teams/TeamsHero.tsx [hunk 9]
```diff
@@ -1,38 +1,30 @@
+import Image, { ImageProps } from "next/image"
 
-function TeamsHero() {
-
+const TeamsHero: React.FC = () => {
+    const images: ImageProps[] = [{src: "/img/teams/team-work/team-work-1.png", alt:"", className:'contain absolute top-0 left-0 w-[35%]' ,width:322,height:316.7}, {src: "/img/teams/team-work/team-work-2.png", alt:"", className: 'contain absolute top-0 right-0 w-[35%]', width:322,height:316.7}, {src:"/img/teams/team-work/team-work-3.png",alt:"",width:322,height:316.7, className:'contain absolute bottom-0 left-0 w-[35%]'}, {src:"/img/teams/hero-background.png", alt:"", className: "",width:800,height:710.7}]
```
Comment (ruthgdb) [inline Web/starter_project_iconics/components/teams/TeamsHero.tsx:4]: This is not readable. Can you format this?

**Review by ruthgdb — CHANGES_REQUESTED** [review]
Looks good overall. Please address this reviews before we merge this.

**Review by ruthgdb — APPROVED** [review]
Good job.

**Pull request merged**